src/test/java/StringManipulationTest.java

```diff
@@ -22,65 +22,88 @@ public void tearDown() {
     @Test
```

Comment:
This is a general comment for all the tests:
I've noticed that you aren't testing for exceptions here. I'm not sure if that's a conscious choice or not on your part, so I won't say you need to. I just think that it's better coding practice to test for your exceptions, make sure that they're being thrown when you want them to be. It would also help with some of the tests being similar to each other.